### PR TITLE
[RzIL] Refactoring, added bool xor and added bitv carry/borrow

### PR DESCRIPTION
**Pull request by wargio**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Small refactoring for names and structures, added boolean xor operation and implemented carry for `add` and borrow for `sub`

### CODE REVIEW

File: librz/analysis/p/analysis_bf.c [hunk 1]
```diff
@@ -274,22 +274,22 @@ RzPVector *bf_rlimit(RzILVM *vm, BfContext *ctx, ut64 id, ut64 addr) {
 	RzILOp *branch = rz_il_new_op(RZIL_OP_BRANCH);
 	RzILOp *goto_ = rz_il_new_op(RZIL_OP_GOTO);
 	RzILOp *perform = rz_il_new_op(RZIL_OP_PERFORM);
-	RzILOp *inv = rz_il_new_op(RZIL_OP_INV);
```
Comment (XVilka) [inline librz/analysis/p/analysis_bf.c]: Why this change? It's defined as `inv` in the original: https://github.com/BinaryAnalysisPlatform/bap/blob/master/lib/bap_core_theory/bap_core_theory_definition.ml#L49
```ocaml

module type Bool = sig
  val b0 : bool
  val b1 : bool
  val inv : bool -> bool
  val and_ : bool -> bool -> bool
  val or_ : bool -> bool -> bool
end
```

File: librz/il/rzil_export.c [hunk 2]
```diff
@@ -187,16 +187,20 @@ static void il_opdmp_b1(RzILOp *op, RzStrBuf *sb, PJ *pj) {
 	}
 }
 
-static void il_opdmp_inv(RzILOp *op, RzStrBuf *sb, PJ *pj) {
-	il_op_param_2("inv", op->op.inv, x, ret);
+static void il_opdmp_bool_not(RzILOp *op, RzStrBuf *sb, PJ *pj) {
```
Comment (XVilka) [inline librz/il/rzil_export.c]: Then this will be `il_opdmp_bool_inv`

File: librz/il/theory_bool.c [hunk 3]
```diff
@@ -4,28 +4,28 @@
...
+void *rz_il_handler_bool_false(RzILVM *vm, RzILOp *op, RzILOpArgType *type) {
 	rz_return_val_if_fail(vm && op && type, NULL);
 
 	RzILBool *ret = rz_il_bool_new(false);
 	*type = RZIL_OP_ARG_BOOL;
 	return ret;
 }
 
-void *rz_il_handler_b1(RzILVM *vm, RzILOp *op, RzILOpArgType *type) {
+void *rz_il_handler_bool_true(RzILVM *vm, RzILOp *op, RzILOpArgType *type) {
```
Comment (XVilka) [inline librz/il/theory_bool.c:21]: Please keep original operation names. If the name is unclear, it's better to add a Doxygen comment explaining it.
Refer to the names in https://github.com/BinaryAnalysisPlatform/bap/blob/master/lib/bap_core_theory/bap_core_theory_definition.ml#L49

File: librz/il/theory_bool.c [hunk 4]
```diff
@@ -4,28 +4,28 @@
 #include <rz_il/rzil_opcodes.h>
 #include <rz_il/rzil_vm.h>
 
-void *rz_il_handler_b0(RzILVM *vm, RzILOp *op, RzILOpArgType *type) {
+void *rz_il_handler_bool_false(RzILVM *vm, RzILOp *op, RzILOpArgType *type) {
```
Comment (XVilka) [inline librz/il/theory_bool.c:10]: Or, we could use these new names but add a comment to which BAP CT operation it corresponds.

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/il/definitions/bitvector.c [hunk 5]
```diff
@@ -554,27 +554,28 @@ RZ_API RZ_OWN RzILBitVector *rz_il_bv_complement_2(RZ_NONNULL RzILBitVector *bv)
  * Result of (x + y) mod 2^length
  * \param x RzILBitVector, Operand
  * \param y RzILBitVector, Operand
```
Comment (thestr4ng3r) [inline librz/il/definitions/bitvector.c:557]: While you are at it, please document that both operands must have the same size.

File: librz/analysis/p/analysis_bf.c (hunk 1 above)
Comment (wargio) [reply librz/analysis/p/analysis_bf.c]: well, the normal op boolean name is not, not inv

**Review by ret2libc — COMMENTED** [review]
LGTM

File: librz/analysis/p/analysis_bf.c (hunk 1 above)
Comment (XVilka) [reply librz/analysis/p/analysis_bf.c]: Well, I know. But I think it's better to stick to the original, as @thestr4ng3r noted too.

Comment (wargio) [reply librz/analysis/p/analysis_bf.c]: i disagree. on bap, names have been choosen sue the limitation of the lang, this is why you have `not_` or `and_` etc..

Comment (wargio) [reply librz/analysis/p/analysis_bf.c]: i'm ok to map them to the original bap names via comments. but the name of the structure, etc.. has to be clear in C

Comment (XVilka) [reply librz/analysis/p/analysis_bf.c]: I am fine with the new names as long as there corresponding comments supplied. BOOLNOT looks weird though

Comment (wargio) [reply librz/analysis/p/analysis_bf.c]: not really, we have logical not and boolean not. in C the difference is `ut32 x = 0x1234` 
*Boolean Not* is `!x` meanwhile *Logical Not* is `~x`

Comment (XVilka) [reply librz/analysis/p/analysis_bf.c]: Yes, I understand the difference, just complaining about naming.

Comment (wargio) [reply librz/analysis/p/analysis_bf.c]: it follows the names for logical and/or/xor

**Comment by XVilka** [comment]
@wargio please add corresponding comments and lets merge this.

File: librz/analysis/p/analysis_bf.c (hunk 1 above)
Comment (thestr4ng3r) [reply librz/analysis/p/analysis_bf.c]: I think we should keep it as `inv`. Our source of reference is bap and not what may be subjectively perceived as more standard.

**Review by XVilka — APPROVED** [review]

**Review by thestr4ng3r — APPROVED** [review]
